cranelift/codegen/src/isa/s390x/inst.isle

```diff
@@ -2473,6 +2482,20 @@
 (rule (emit_load $I64 dst addr)
       (emit (MInst.Load64 dst addr)))
 
+;; Helper for creating `MInst.MovPReg` instructions.
+(decl mov_preg (PReg) Reg)
+(rule (mov_preg src)
+      (let ((dst WritableReg (temp_writable_reg $I64))
+            (_ Unit (emit (MInst.MovPReg dst src))))
+        dst))
+
+(decl preg_r15 () PReg)
+(extern constructor preg_r15 preg_r15)
```

Comment:
Just a suggestion to use something like `preg_stack` instead of r15, since this is actually not an ISA requirement - other ABIs use different registers as stack pointer.